### ISSUE TITLE
May something wrong with concealcursor.

### ISSUE THREAD

**Issue by yangxyo** [vim/vim#6375]
I want to hide the char of the first-level tab.
```
set listchars=tab:\¦\ ,trail:■,extends:>,precedes:<,nbsp:+
set list
hi SpecialKey ctermfg=239 ctermbg=202

set conceallevel=1
set concealcursor=inc
call matchadd('Conceal', '^\t', 0, -1, {'conceal':'	'})
hi Conceal ctermfg=NONE ctermbg=NONE
```
![Screenshots](https://user-images.githubusercontent.com/32320149/86306248-55b6d700-bc46-11ea-8a0b-39868f7df76d.gif)

```
VIM - Vi IMproved 8.2
MacOS 10.15.5 
```
I don't know whether it is a bug. So I created a blank one.

**Comment by k-takata** [comment]
Could you explain what is your expected behavior?

**Comment by yangxyo** [comment]
```
set listchars=tab:\¦\ ,trail:■,extends:>,precedes:<,nbsp:+
set list
hi SpecialKey ctermfg=239 ctermbg=202
```
It started like this:
<img src="https://user-images.githubusercontent.com/32320149/86309990-9830e180-bc4f-11ea-89eb-86e1cd658cce.png" width="400px" alt="">

But I woloud like this(Do not show the first-level tab char):
<img src="https://user-images.githubusercontent.com/32320149/86310020-a5e66700-bc4f-11ea-9f53-5c5c0dee7d05.png" width="400px" alt="">

So I add:
```
set conceallevel=1
set concealcursor=inc
call matchadd('Conceal', '^\t', 0, -1, {'conceal':'	'})
hi Conceal ctermfg=NONE ctermbg=NONE
```

But actually show(When I move the cursor the indent will change, it's strange):
<img src="https://user-images.githubusercontent.com/32320149/86310042-b0086580-bc4f-11ea-92f2-75e6bfbbe672.png" width="400px" alt="">

**Comment by yangxyo** [comment]
> Could you explain what is your expected behavior?

Is there any way to solve it？

**Comment by chrisbra** [comment]
While I think this is already stretching what an editor should do a bit too much by using concealing and listchars, it looks like a bug. I see rendering issues on the line where the cursor is (displaying different characters). In addition, it looks like the concealing code does not handle using a tab as a conceal character very well.

**Comment by chrisbra** [comment]
reproduced with 8.2.1114 using the following snippet:

```
set listchars=tab:\¦\ ,trail:■,extends:>,precedes:<,nbsp:+
set list ts=8
hi SpecialKey ctermfg=239 ctermbg=202
set conceallevel=1
set concealcursor=inc
call matchadd('Conceal', '^\t', 0, -1, {'conceal':"\t"})
hi Conceal ctermfg=NONE ctermbg=NONE

	"foobar eins 	zwei
	"foobar eins 	zwei
	"foobar eins 	zwei

```

Note how the line where the cursor is is rendered differently (wrong). See commandline for the actual content.
![grafik](https://user-images.githubusercontent.com/244927/86322802-e936da80-bc3b-11ea-87e8-7a9fb77ac8b8.png)

**Comment by k-takata** [comment]
`:help matchadd()` says:

```
			conceal	    Special character to show instead of the
				    match (only for |hl-Conceal| highlighted
				    matches, see |:syn-cchar|)
```

and `:help :syn-cchar` says:

```
The character cannot be a control character such as Tab.
```

So, this is a limitation?

**Comment by chrisbra** [comment]
Oh yes, indeed. I remember having read this in the past, but missed it.

**Comment by yangxyo** [comment]
> and `:help :syn-cchar` says:
> 
> ```
> The character cannot be a control character such as Tab.
> ```
> 
> So, this is a limitation?

Because of this, I use `{'conceal':'	'})`, rather than `{'conceal':'	\t'})`
And I do realize it.
As the second image:
https://github.com/vim/vim/issues/6375#issuecomment-652744851
However, as the third image, when my cursor passes the line, it crashes.

**Comment by chrisbra** [comment]
> Because of this, I use {'conceal':' '}), rather than {'conceal':' \t'})

That was not in your original issue. It contained a literal tab, I just converted it to use `\t` instead.

> However, as the third image, when my cursor passes the line, it crash.

What do you mean here? Can you post a backtrace please?

**Comment by yangxyo** [comment]
![image](https://user-images.githubusercontent.com/32320149/86326887-ec9a8800-bc74-11ea-83cd-0d00a429c987.png)

**Comment by yangxyo** [comment]
It seems `set concealcursor=inc`, its performance doesn't like the `help: concealcursor`.

**Comment by yangxyo** [comment]
😂But，in fact, I don't want to feedback bug, I'm looking for the way to "hide the character of the first-level tab".

**Comment by yangxyo** [comment]
It not only the indent unusual, but it also completes collapse.
![QQ20200702-151432-HD 2020-07-02 15_15_18](https://user-images.githubusercontent.com/32320149/86328107-dc83a800-bc76-11ea-9611-93961d5b3971.gif)
I only type the "j->o -> fo -> dd -> dsfaf->k"

**Comment by yangxyo** [comment]
@chrisbra 
> Because of this, I use {'conceal':' '}), rather than {'conceal':' \t'})
> 
> That was not in your original issue. It contained a literal tab, I just converted it to use `\t` instead.

I first don't think:  ` {'conceal':'	'}` ==  `{'conceal':' \t'}` , so I use  `{'conceal':'	'} `😂.

However, I still think it is sth. wrong because it worked first, it is that the `set concealcursor=inc` does not work as far as I see, hah.

And I only need sth. to replace or hide the `¦` of `'^\t'`. However, I don't allow to use four spaces`    `, because it is `cstr` that can't be used for `matchadd` of Vim now, it will be analyzed to one space.

**Comment by yangxyo** [comment]
@chrisbra 
> While I think this is already stretching what an editor should do a bit too much by using concealing and listchars

Is there a way to` set listchars =tab:\ \ ` for the first-level tab, while `set listchars =tab:\¦\ ` for others?

**Comment by chrisbra** [comment]
> Is there a way toset listchars =tab:\ \ for the first-level tab, while set listchars =tab:\¦\ for others?

No, listchars is intended to be used for all characters that follow. Perhaps you should try to explain, what actual problem you are trying to solve. I must confess, I couldn't follow your messages and it is not clear to me what you want to achieve and why.

**Comment by yangxyo** [comment]
I'm so sorry.

> What actual problem you are trying to solve

I need the Indent alignment lines.
```
set listchars=tab:\¦\ ,trail:■,extends:>,precedes:<,nbsp:+
set list
hi SpecialKey ctermfg=239 ctermbg=202
```
It started like this:
<img src="https://user-images.githubusercontent.com/32320149/86309990-9830e180-bc4f-11ea-89eb-86e1cd658cce.png" width="400px" alt="">

But I would like this **which does not display the first-level tab character**:
<img src="https://user-images.githubusercontent.com/32320149/86310020-a5e66700-bc4f-11ea-9f53-5c5c0dee7d05.png" width="400px" alt="">

> Why

I think It will be more reasonable, simple, and beautiful. "Radish or cabbage; each to his own delight."
I hope I explained it clearly.🌚
